spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/RewriteDataFilesSparkAction.java

```diff
@@ -489,7 +490,10 @@ private String jobDesc(RewriteFileGroup group, RewriteExecutionContext ctx) {
   }
 
   private BinPackStrategy binPackStrategy() {
-    return new SparkBinPackStrategy(table, spark());
+    // Disable Adaptive Query Execution as this may change the output partitioning of our write
+    SparkSession spark = spark().cloneSession();
+    spark.conf().set(SQLConf.ADAPTIVE_EXECUTION_ENABLED().key(), false);
+    return new SparkBinPackStrategy(table, spark);
```

Comment:
Bin packing is used as the basis for the other strategies. If you're moving this into the action itself, then you should move it into `spark()`.

Reply:
If it's moved into `spark()` then SparkSession is cloned with AQE disabled for all `BaseSparkAction`s. I'm not sure about the side-effect.

Reply:
Given that `spark()` is protected in `BaseSparkAction`, it could be overridden within this action.

Or another `spark()` method could be added that clones and disables AQE, like `spark(boolean cloneAndDisableAQE)` or something.

Reply:
I don't see the difference here since we need to invoke the `spark()` from super class first.